src/m365/flow/commands/owner/owner-add.ts

```diff
@@ -0,0 +1,179 @@
+      },
+      {
+        option: '--roleName <roleName>',
+        autocomplete: FlowOwnerAddCommand.allowedRoleNames
+      },
+      {
+        option: '--asAdmin'
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['userId', 'userName', 'groupId', 'groupName'] });
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (!validation.isValidGuid(args.options.flowName)) {
+          return `${args.options.flowName} is not a valid GUID.`;
+        }
+
+        if (args.options.userId && !validation.isValidGuid(args.options.userId)) {
+          return `${args.options.userId} is not a valid GUID.`;
+        }
+
+        if (args.options.userName && !validation.isValidUserPrincipalName(args.options.userName)) {
+          return `${args.options.userName} is not a valid userName.`;
+        }
+
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId)) {
+          return `${args.options.groupId} is not a valid GUID.`;
+        }
+
+        if (FlowOwnerAddCommand.allowedRoleNames.indexOf(args.options.roleName) < 0) {
+          return `${args.options.roleName} is not a valid roleName. Valid values are ${FlowOwnerAddCommand.allowedRoleNames.join(', ')}`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      if (this.verbose) {
+        logger.logToStderr(`Assigning permissions for ${args.options.userId || args.options.userName || args.options.groupId || args.options.groupName} with permissions ${args.options.roleName} to Power Automate flow ${args.options.flowName}`);
+      }
+
+      let id = '';
+      if (args.options.userId) {
+        id = args.options.userId;
+      }
+      else if (args.options.userName) {
+        id = await aadUser.getUserIdByUpn(args.options.userName);
+      }
+      else if (args.options.groupId) {
+        id = args.options.groupId;
+      }
+      else {
+        const resp = await aadGroup.getGroupByDisplayName(args.options.groupName!);
+        id = resp.id!;
+      }
+
+      let type: string = '';
+      if (args.options.userId || args.options.userName) {
+        type = 'User';
+      }
+      else {
+        type = 'Group';
+      }
+
+      const requestOptions: CliRequestOptions = {
+        url: `${this.resource}providers/Microsoft.ProcessSimple/${args.options.asAdmin ? 'scopes/admin/' : ''}environments/${formatting.encodeQueryParameter(args.options.environmentName)}/flows/${formatting.encodeQueryParameter(args.options.flowName)}/modifyPermissions?api-version=2016-11-01`,
+        headers: {
+          accept: 'application/json'
+        },
+        data: {
+          put: [
+            {
+              properties: {
+                principal: {
+                  id: id
+                },
+                roleName: args.options.roleName
+              },
+              type: type
```

Comment:
Maybe my last comment was not clear enough, sorry for that!
The property `type` is part of the `principal` object. If we don't put it there, the output while retrieving the owner looks like:

```json
{
"properties": {
      "roleName": "CanEdit",
      "permissionType": "Principal",
      "principal": {
        "id": "91ff2e17-84de-455a-8815-52b21683f64e",
        "type": "NotSpecified"
      }
    }
}
```

We should add `User` or `Group` instead of `NotSpecified`.

Reply:
No problem, my bad! Updated the PR